content-services/content-sources-backend
# Fixes 4927: add flag to prevent requeuing canceled tasks

## Summary

Adds a `cancel_attempted` field to the tasks table and prevents requeuing a task if it was canceled

## Testing steps

1. On main, create a repo and let it snapshot
2. You can use this script to reproduce the "could not find template with UUID" errors (you'd need to adjust it to use your identity header and change the `repository_uuids` to include the repo you've created). Running this on main should reproduce those errors (you may not need to loop as many times):

```
for i in {1..50}; do
  UUID=$(curl -s -X POST --location "http://localhost:8000/api/content-sources/v1.0/templates/"     -H "x-rh-identity: eyJpZGVudGl0eSI6eyJvcmdfaWQiOiIxNzY4NDYzMiIsICJ0eXBlIjoiVXNlciIsInVzZXIiOnsidXNlcm5hbWUiOiJicnl0dGFuaWVob3VzZSJ9LCJhY2NvdW50X251bWJlciI6IjExNDkxNzg2IiwiaW50ZXJuYWwiOnsib3JnX2lkIjoiMTc2ODQ2MzIifX19Cg=="     -H "Content-Type: application/json"     -d "{
          \"name\":\"test-$RANDOM\",
          \"arch\": \"x86_64\",
          \"version\": \"8\",
          \"repository_uuids\": [\"0de58827-5301-44da-a226-85db660d6d1f\"],
          \"date\": \"2024-11-05T14:04:05Z\"
        }"  | jq -r .uuid)

  curl -s -X DELETE --location "http://localhost:8000/api/content-sources/v1.0/templates/$UUID"     -H "x-rh-identity: eyJpZGVudGl0eSI6eyJvcmdfaWQiOiIxNzY4NDYzMiIsICJ0eXBlIjoiVXNlciIsInVzZXIiOnsidXNlcm5hbWUiOiJicnl0dGFuaWVob3VzZSJ9LCJhY2NvdW50X251bWJlciI6IjExNDkxNzg2IiwiaW50ZXJuYWwiOnsib3JnX2lkIjoiMTc2ODQ2MzIifX19Cg=="

done
```
3. Checkout this PR, run the migration, and run the script again. You shouldn't see any requeues or "could not find template with UUID" errors
